data/dataset.py

```diff
@@ -0,0 +1,98 @@
+import random
+
+import torch
+import numpy as np
+from torch.utils.data import Dataset
+from datasets import Dataset as HFDataset
+
+from data.masking import AwesomeMasks
+from data.tokenizer import MidiEncoder
+from data.quantizer import MidiQuantizer
+from data.augmentation import pitch_shift, change_speed
+
+
+class MidiDataset(Dataset):
+    def __init__(
+        self, dataset: HFDataset, tokenizer: MidiEncoder, augmentation_probability: float = 0.0, masking_probability: float = 0.15
+    ):
+        super().__init__()
+
+        self.dataset = dataset
+        self.quantizer = MidiQuantizer(7, 7, 7)
```

Comment:
```suggestion
    def __init__(
        self,
        dataset: HFDataset, 
        tokenizer: MidiEncoder, 
        augmentation_probability: float = 0.0, 
        masking_probability: float = ???,
        quantizer: MidiQuantizer,
    ):
        super().__init__()

        self.dataset = dataset
        self.quantizer = quantizer
```

Let's not hardcode quantization parameters :)

With the current setup, would it be difficult to make the masking probability dynamic? I'm not sure how this could work exactly, but we may define a range, let's say `[0.1, 0.5]` and maybe treat different values as augmentation 🤔

Reply:
Actually, with the `masking_space` approach, it should be possible to define masking probability during train-time, so we should be able to remove it from the dataset class completely?